skt/runner.py

```diff
@@ -460,21 +461,18 @@ def run(self, url, release, wait=False, host=None, uid="",
         return ret
 
 
-def getrunner(rtype, rarg):
-    """
-    Create an instance of a "runner" subclass with specified arguments.
-
-    Args:
-        rtype:  The value of the class "TYPE" member to match.
-        rarg:   A dictionary with the instance creation arguments.
-
-    Returns:
-        The created class instance.
-
-    Raises:
-        ValueError if the rtype match wasn't found.
-    """
-    for cls in Runner.__subclasses__():
-        if cls.TYPE == rtype:
-            return cls(**rarg)
-    raise ValueError("Unknown runner type: %s" % rtype)
+def getrunner(cfg):
+    """Create an instance of a "runner" subclass with specified arguments."""
+    try:
+        module = importlib.import_module('skt.runner')
+        class_name = "{}Runner".format(cfg['runner']['type'].title())
+        reporter_class = getattr(module, class_name)
+    except AttributeError:
+        sys.exit(
+            "Unable to find specified runner type: {}".format(class_name)
+        )
+
+    return reporter_class(
+        jobtemplate=cfg['runner']['jobtemplate'],
+        jobowner=cfg['runner']['jobowner']
+    )
```

Comment:
Could you please make this refactoring a separate commit, and also please keep documenting both arguments and return values?

Reply:
Actually, I'm not sure this is a good idea. It makes the class naming tied to the external interface and would make code refactoring difficult while keeping compatibility. Also, this is way too fancy/complicated. I would prefer to just have an explicit, simple dictionary of runner names vs types (classes).

Also, please don't forget to remove the "TYPE" variable from the runner class if you're going to do this in any shape.

Reply:
But the class naming was tied to the runner type passed via `cfg` already. Are you suggesting we make a dictionary to map a runner name to a particular class?

Reply:
Hmm, I think it was tied to the "TYPE" member value. Yeah, I was thinking we make a dictionary.

And actually I think we shouldn't have it in the runner module, but instead in the executable.py, where the configuration/options are being interpreted. And the same for all other classes chosen by configuration/options. That can be a separate PR, if we decide to do it.

Reply:
So something like this?

```python
runner_types = { 'beaker': 'BeakerRunner', 'something_else': "OtherRunner" }
```

And just instantiate whatever class name is returned from `runner_types['beaker']`?

Reply:
Okay, I backed out some of the refactoring of `runner.getrunner()` and I'll work that dictionary stuff in another PR.